src/libasr/codegen/llvm_utils.cpp

```diff
@@ -295,6 +295,47 @@ namespace LFortran {
         shift_end_point_by_one(list);
     }
 
+    void LLVMList::insert_item(llvm::Value* list, llvm::Value* pos,
+                               llvm::Value* item, llvm::Module& module,
+                               std::string& type_code) {
+        llvm::Value* current_end_point = LLVM::CreateLoad(*builder, get_pointer_to_current_end_point(list));
+        llvm::Value* current_capacity = LLVM::CreateLoad(*builder, get_pointer_to_current_capacity(list));
+        int type_size = std::get<1>(typecode2listtype[type_code]);
+        llvm::Type* el_type = std::get<2>(typecode2listtype[type_code]);
+        resize_if_needed(list, current_end_point, current_capacity,
+                         type_size, el_type, module);
+
+        llvm::Value* end_point = LLVM::CreateLoad(*builder, get_pointer_to_current_end_point(list));
+        llvm::Value* tmp1 = LLVM::CreateLoad(*builder, read_item(list, pos, true));
+        llvm::Value* tmp2 = nullptr;
+        llvm::Value* next_element_ptr = llvm::ConstantInt::get(context, llvm::APInt(32, 1));
+        llvm::Value* copy_pos = pos;
+
+
+        llvm::BasicBlock *loophead = llvm::BasicBlock::Create(context, "loop.head");
+        llvm::BasicBlock *loopbody = llvm::BasicBlock::Create(context, "loop.body");
+        llvm::BasicBlock *loopend = llvm::BasicBlock::Create(context, "loop.end");
+
+        // head
+        llvm_utils->start_new_block(loophead);
+        llvm::Value *cond = builder->CreateICmpSLT(pos, end_point);
+        builder->CreateCondBr(cond, loopbody, loopend);
+
+        // body
+        llvm_utils->start_new_block(loopbody);
+        next_element_ptr = builder->CreateAdd(copy_pos, next_element_ptr);
+        tmp2 = LLVM::CreateLoad(*builder, read_item(list, next_element_ptr, true));
+        write_item(list, next_element_ptr, tmp1);
+        tmp1 = tmp2;
+        pos = builder->CreateAdd(pos, llvm::ConstantInt::get(context, llvm::APInt(32, 1)));
+        builder->CreateBr(loophead);
```

Comment:
I tried to debug here, 
Initially `pos` is `1` as per the example.
`pos` incremented by `1` resulting in `2` in the body, but when it returns to the loop head, `pos` has the same value i.e., `1`